File: src/main/java/io/dddrive/search/model/SearchResult.java

package io.dddrive.search.model;

import java.math.BigDecimal;

import io.dddrive.ddd.model.enums.CodeAggregateType;
import io.dddrive.oe.model.ObjTenant;

public class SearchResult implements Comparable<SearchResult> {

	private ObjTenant tenant;
	private CodeAggregateType aggregateType;
	private Integer id;
	private String caption;
	private BigDecimal rank;

	public SearchResult(ObjTenant tenant, CodeAggregateType aggregateType, Integer id, String caption, BigDecimal rank) {
		this.tenant = tenant;
		this.aggregateType = aggregateType;
		this.id = id;
		this.caption = caption;
		this.rank = rank;
	}

	public ObjTenant getTenant() {
		return this.tenant;
	}

	public CodeAggregateType getAggregateType() {
		return this.aggregateType;
	}

	public String getId() {
		return String.valueOf(this.id);
	}

	public String getCaption() {
		return this.caption;
	}

	public BigDecimal getRank() {
		return this.rank;
	}

	@Override
	public int compareTo(SearchResult other) {
		return this.rank.compareTo(other.getRank());
	}

}

File: fm-domain/src/main/java/io/zeitwert/dddrive/ddd/model/SearchResult.kt
package io.zeitwert.dddrive.ddd.model

import dddrive.ddd.model.enums.CodeAggregateType
import java.math.BigDecimal

data class SearchResult(
	val tenantId: Int,
	val aggregateType: CodeAggregateType,
	val id: Int,
	val caption: String,
	val rank: BigDecimal,
) : Comparable<SearchResult> {

	override fun compareTo(other: SearchResult): Int = this.rank.compareTo(other.rank)

}
